modules/core/src/main/scala/org/scalasteward/core/data/Version.scala

```diff
@@ -91,7 +87,7 @@ final case class Version(value: String) {
   private def containsHash: Boolean =
     components.exists {
       case _: Version.Component.Hash => true
-      case _                         => false
+      case _ => Rfc5234.hexdig.rep(8).string.filterNot(startsWithDate).parse(value).isRight
     }
```

Comment:
If the version contains no hash this will check for hashes in `value` on every component. How about moving this after the `exists` call:
```scala
    components.exists {
      case _: Version.Component.Hash => true
      case _                         => false
    } || Rfc5234.hexdig.rep(8).string.filterNot(startsWithDate).parse(value).isRight
```
?

Reply:
makes sense 👍